modules/googleplay/src/main/scala/cards/nine/googleplay/service/free/interpreter/cache/CacheInterpreter.scala

```diff
@@ -37,48 +36,52 @@ class CacheInterpreter(implicit ec: ExecutionContext) extends (Ops ~> RedisOps)
 
   private[this] val pendingSet: CacheSet[PendingQueueKey.type, Package] = new CacheSet(PendingQueueKey)
 
-  def apply[A](ops: Ops[A]): RedisOps[A] = ops match {
-
-    case GetValid(pack) ⇒
-      val keys = List(CacheKey.resolved(pack), CacheKey.permanent(pack))
-      wrap.mget(keys).map(_.flatMap(_.card).headOption)
-
-    case GetValidMany(packages) ⇒
-      val keys = (packages map CacheKey.resolved) ++ (packages map CacheKey.permanent)
-      wrap.mget(keys).map(_.flatMap(_.card))
+  override def getValid(pack: Package): RedisOps[Option[FullCard]] = {
+    val keys = List(CacheKey.resolved(pack), CacheKey.permanent(pack))
+    wrap.mget(keys).map(_.flatMap(_.card).headOption)
+  }
 
-    case PutResolved(card) ⇒
-      val pack = card.packageName
-      pendingSet.remove(pack) *> removeError(pack) *> wrap.put(CacheEntry.resolved(card))
+  override def getValidMany(packages: List[Package]) = {
```

Comment:
I've seen that checked before with wart remover but not sure of the compiler implements that too in some flag